### ISSUE TITLE
can whatsat handle group chat?

### ISSUE THREAD

**Issue by philbertw4** [joostjager/whatsat#5]
This is not a bug report, but more just a question and possible discussion.

Whatsat is an interesting use case for lightning network! Thank you for creating it. I wonder if it might be possible to modify the whatsat protocol such that it can allow multi-node chatrooms? If so, then it seems that perhaps we could also build bots/services which could also be quite useful (e.g. imagine https://telegram.me/lntxbot with all of its plugins and tools being available to whatsat users). 

What do you think? Will such a thing (eventually) be possible?

**Comment by joostjager** [comment]
Yes, it is an interested question. I am not sure how other messaging apps technically handle group chats. If there is a central server, a single message can be fanned out to all group participants I'd imagine. 

For chat over lightning, I think it could work by having the sender of the message send it to all participants. It is not very efficient though with a big group, but maybe that turns out to not matter much.

Also participants would need to know who the other participants are. If one person is the owner of the chat room, they could broadcast a message to all others when someone new joins.

**Comment by philbertw4** [comment]
Yeah, I suppose there are merits to both methods:
1. sender sending to all participants
2. a specific node hosting a chatroom and rebroadcasting
I too am not sure how other messaging apps handle this, but it is definitely an interesting thought and would be a very useful feature. I need to play with whatsat myself first to see how it actually works. Any chance it'll be on mainnet soon?

**Comment by joostjager** [comment]
You can run it on mainnet if you want to, but just be careful.

**Comment by joostjager** [comment]
Related: https://eprint.iacr.org/2019/1416

**Comment by jbostick00** [comment]
Very cool project, awesome to see these ideas coming to life.
